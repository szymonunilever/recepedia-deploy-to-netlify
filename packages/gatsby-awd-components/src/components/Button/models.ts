import { ReactNode } from 'react';
import { Icon } from '../../models';

export interface ButtonCallback {
  // @ts-ignore
  (...arg): void;
}
export enum ButtonViewType {
  classic,
  icon,
}

export interface ButtonProps {
  className?: string;
  Icon?: Icon;
  IconSelected?: Icon;
  content?: AppContent.CTAContent;
  role?: string;
  isDisabled?: boolean;
  isSelected?: boolean;
  onClick?: ButtonCallback;
  onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hidden?: boolean;
  isToggle?: boolean;
  type?: 'button' | 'submit' | 'reset';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  attributes?: Record<string, any>;
}
