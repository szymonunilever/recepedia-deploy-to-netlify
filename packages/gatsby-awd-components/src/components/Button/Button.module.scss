@import '../../scss/config';

.button {
  @extend %button;

  &--icon {
    cursor: pointer;
    background: transparent;
    border: 0 none;
    font-size: inherit;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &[disabled] {
    pointer-events: none;
    background-color: $button-disabled-background;
    color: $button-disabled-color;
    cursor: default;

    @media (hover: hover),
      (-ms-high-contrast: none),
      (-ms-high-contrast: active) {
      &:hover {
        background-color: $button-disabled-background !important;
        color: $button-disabled-color !important;
      }
    }
  }
  // Button color modification
}
:global {
  .button {
    &--secondary {
      background-color: $button-secondary-background;
      color: $button-secondary-color;
      font-weight: normal;
    }
    &--large {
      width: 100%;
      max-width: $button-large-max-width;
    }

    &--medium {
      width: 100%;
      max-width: $button-medium-max-width;
      @include respondTo(desktop) {
        max-width: $button-medium-max-width-lg;
      }
    }

    &--small {
      width: 100%;
      max-width: $button-small-max-width;
      @include respondTo(desktop) {
        max-width: $button-small-max-width-lg;
      }
    }
  }
}
