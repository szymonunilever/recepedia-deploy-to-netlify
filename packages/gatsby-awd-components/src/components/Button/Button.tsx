/* eslint-disable @typescript-eslint/camelcase */
import cx from 'classnames';
import React, { FunctionComponent, MouseEvent, useEffect, useState } from 'react';
import theme from './Button.module.scss';
import { ButtonProps, ButtonViewType } from './models';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';
import { iconNormalize } from '../../utils';

export const Button: FunctionComponent<ButtonProps> = ({
  Icon,
  IconSelected = Icon,
  isDisabled = false,
  isSelected = false,
  onClick,
  onKeyDown,
  hidden = false,
  className = '',
  children = '',
  type = 'button',
  isToggle = false,
  viewType = ButtonViewType.classic,
  content,
  role,
  attributes = {},
}) => {
  const { label } = content || { label: '' };
  const [selected, setSelected] = useState<boolean | undefined>(isSelected);
  useEffect(() => {
    setSelected(isSelected);
  }, [isSelected]);
  const buttonClasses = cx(
    viewType === ButtonViewType.icon
      ? cx(theme.buttonIcon, 'button--icon')
      : cx(theme.button, 'button'),
    className,
    {
      button__selected: selected,
      button__disabled: isDisabled,
    }
  );

  const IconComponent = selected
    ? iconNormalize(IconSelected)
    : iconNormalize(Icon);
  const onButtonClick = (e: MouseEvent) => {
    type === 'button' && e.preventDefault();

    if (isToggle) {
      const newVal = !selected;
      setSelected(newVal);
      onClick && onClick(newVal);
    } else {
      onClick && onClick(false);
    }
  };
  const props = {
    role,
    type,
    ...getComponentDataAttrs('button', content),
    onClick: onButtonClick,
    disabled: isDisabled,
    hidden,
  };

  if (role === 'tab') {
    // @ts-ignore
    isToggle ? (props['aria-selected'] = selected) : null;
  } else {
    // @ts-ignore
    isToggle ? (props['aria-pressed'] = selected) : null;
  }

  return (
    <button
      className={buttonClasses}
      {...props}
      {...attributes}
      onKeyDown={onKeyDown}
    >
      {IconComponent}
      {label ? <span className="button__label">{label}</span> : null}
      {children}
    </button>
  );
};

export default Button;
