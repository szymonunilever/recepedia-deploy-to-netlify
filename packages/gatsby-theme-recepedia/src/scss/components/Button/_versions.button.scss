@import '../../../../../gatsby-awd-components/src/scss/config'; // way of including library config may be reviewed later

.button {
  @extend %button;

  &--rounded {
    border-radius: 20px;
  }

  &--circle {
    border-radius: 50%;
  }

  // Button color modification
  &--secondary {
    background-color: $button-secondary-background;
    color: $button-secondary-color;
    font-weight: normal;
  }

  &--yellow {
    background-color: $luminous-yellow;
    color: $black-blue;
    fill: currentColor;

    @media (hover: hover),
      (-ms-high-contrast: none),
      (-ms-high-contrast: active) {
      &:hover {
        background-color: $button-background-hover;
        color: $white;
      }
    }
  }

  // Button size modification

  &--fullwidth {
    width: 100%;
    max-width: $button-large-max-width;
  }

  &--medium {
    width: 100%;
    max-width: $button-medium-max-width;
    @include respondTo(desktop) {
      max-width: $button-medium-max-width-lg;
    }
  }

  &--small {
    width: 100%;
    max-width: $button-small-max-width;
    @include respondTo(desktop) {
      max-width: $button-small-max-width-lg;
    }
  }

  // Button With icon

  &--control {
    padding: 0;
    width: 46px;
    height: 46px;
    @include respondTo(desktop) {
      width: 56px;
      height: 56px;
    }
  }

  &--tooltip {
    padding: 0;
    width: 46px;
    height: 46px;
  }

  &--aside {
    padding: 0;
    width: 46px;
    height: 46px;
    @include respondTo(desktop) {
      width: 74px;
      height: 74px;
    }
  }

  &--filter {
    //advanced recipe listing
    background-color: $black-blue;
    width: 40px;
    height: 40px;
    font-size: 1.5rem;
    border-radius: 0;
    padding: 0;
    svg {
      fill: $white;
    }
  }

  &__underlined {
    &:before {
      content: '';
      position: absolute;
      left: 0;
      bottom: 0;
      width: 0;
      height: 1px;
      line-height: 1px;
      background: $black-blue;
      transition: width 0.2s ease-out;
    }
    @media (hover: hover),
      (-ms-high-contrast: none),
      (-ms-high-contrast: active) {
      &:hover {
        &:before {
          width: 100%;
        }
      }
    }
  }
}

.action-button {
  @extend %button;
  @extend .button--circle;
  padding: 0;
  background: $white;
  color: $fish-blue-darker;
  border: 1px solid $fish-blue-darker;
  @media (hover: hover),
    (-ms-high-contrast: none),
    (-ms-high-contrast: active) {
    &:hover {
      background-color: $apple-green;
      color: $black-blue;
      border: 1px solid $black-blue;
      svg {
        fill: $black-blue;
        path {
          fill: $black-blue;
        }
      }
    }
  }

  width: 48px;
  height: 48px;
  svg {
    stroke: currentColor;
    fill: $white;
    transition: stroke 0.2s ease-out;
    width: 20px;
    height: 18px;
  }
  &.button__selected {
    background-color: $apple-green;
    color: $black-blue;
    svg {
      fill: $black-blue;
    }
  }
}
