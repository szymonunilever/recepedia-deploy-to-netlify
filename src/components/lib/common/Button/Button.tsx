import cx from 'classnames';
import React, { MouseEvent, useState } from 'react';
import { ButtonProps, ButtonViewType } from './models';
import theme from './Button.module.scss';

const Button = ({
  icon,
  iconSelected = icon,
  isSelected = false,
  onClick,
  hidden = false,
  className = '',
  children = '',
  isToggle = false,
  viewType = ButtonViewType.classic,
  label = '',
}: ButtonProps) => {
  const [selected, setSelected] = useState<boolean | undefined>(isSelected);
  const wrapClasses = cx(className, {
    // eslint-disable-next-line @typescript-eslint/camelcase
    button__selected: selected,
  });
  const iconClasses = cx(theme.icon, className, {
    // eslint-disable-next-line @typescript-eslint/camelcase
    button__selected: selected,
  });
  const theIcon = selected ? iconSelected : icon;
  const onButtonClick = (e: MouseEvent) => {
    e.preventDefault();
    if (isToggle) {
      const newVal = !selected;
      setSelected(newVal);
      if (typeof onClick !== 'undefined') {
        onClick(newVal);
      }
    } else {
      if (typeof onClick !== 'undefined') {
        onClick(false);
      }
    }
  };

  const view =
    viewType === ButtonViewType.icon ? (
      <button
        type="button"
        data-componentname="button"
        className={iconClasses}
        onClick={onButtonClick}
        hidden={hidden}
      >
        {theIcon}
      </button>
    ) : (
      <button
        type="button"
        data-componentname="button"
        className={wrapClasses}
        onClick={onButtonClick}
        hidden={hidden}
      >
        {theIcon}
        {label ? <span className="button__label">{label}</span> : null}
        {children}
      </button>
    );

  return <>{view}</>;
};

export default Button;
