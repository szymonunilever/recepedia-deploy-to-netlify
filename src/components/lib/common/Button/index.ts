export { default } from './Button';
export { Button } from './Button';
export * from './models';
