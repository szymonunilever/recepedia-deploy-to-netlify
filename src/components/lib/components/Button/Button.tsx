/* eslint-disable @typescript-eslint/camelcase */
import cx from 'classnames';
import React, { MouseEvent, useEffect, useState } from 'react';
import theme from './Button.module.scss';
import { ButtonProps, ButtonViewType } from './models';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';

export const Button = ({
  Icon,
  IconSelected = Icon,
  isDisabled = false,
  isSelected = false,
  onClick,
  hidden = false,
  className = '',
  children = '',
  type = 'button',
  isToggle = false,
  viewType = ButtonViewType.classic,
  content,
  role,
  attributes = {},
}: ButtonProps) => {
  const { label } = content || { label: '' };
  const [selected, setSelected] = useState<boolean | undefined>(isSelected);
  useEffect(() => {
    setSelected(isSelected);
  }, [isSelected]);
  const buttonClasses = cx(
    viewType === ButtonViewType.icon ? theme.icon : theme.button,
    className,
    {
      button__selected: selected,
      button__disabled: isDisabled,
    }
  );

  const IconComponent = selected ? IconSelected : Icon;
  const onButtonClick = (e: MouseEvent) => {
    e.preventDefault();

    if (isToggle) {
      const newVal = !selected;
      setSelected(newVal);
      onClick && onClick(newVal);
    } else {
      onClick && onClick(false);
    }
  };
  const props = {
    role,
    type,
    ...getComponentDataAttrs('button', content),
    onClick: onButtonClick,
    disabled: isDisabled,
    hidden,
  };

  if (role === 'tab') {
    // @ts-ignore
    isToggle ? (props['aria-selected'] = selected) : null;
  } else {
    // @ts-ignore
    isToggle ? (props['aria-pressed'] = selected) : null;
  }

  return (
    <button className={buttonClasses} {...props} {...attributes}>
      {IconComponent && <IconComponent />}
      {label ? <span className="button__label">{label}</span> : null}
      {children}
    </button>
  );
};

export default Button;
