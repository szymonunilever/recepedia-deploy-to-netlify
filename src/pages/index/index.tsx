import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
