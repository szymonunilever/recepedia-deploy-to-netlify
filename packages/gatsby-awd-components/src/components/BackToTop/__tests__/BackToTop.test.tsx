import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
/** Attention: now "src" is webpack alias for the folder "src/components/lib/stories" for storybook app.
 * It was done for compatibility with site app.
 * So for storybook "src" - it's  "src/components/lib/stories" and for gatsby site src it's "src" folder*/
import { ReactComponent as Icon } from 'src/svgs/inline/plus.svg';
import { ButtonViewType } from '../../Button';
import { BackToTop, BackToTopProps } from '../index';
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';

describe('Back to top', () => {
  let wrapper: ReactWrapper;
  const page = {
    height: '200vh',
    fontSize: '2rem',
  };
  const content: AppContent.BackToTopContent = {
    label: 'Back to top',
  };
  const props: BackToTopProps = {
    content,
    viewType: ButtonViewType.icon,
    Icon,
  };

  window.scroll = jest.fn();

  beforeEach(() => {
    wrapper = mount(
      <div style={page}>
        <BackToTop {...props} />
      </div>
    );
    document.documentElement.scrollTop = 800;
    act(() => {
      window.dispatchEvent(new Event('scroll'));
    });
    wrapper.update();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should scroll top on click', () => {
    const scrollSpy = jest.spyOn(window, 'scroll');
    wrapper.find('button').simulate('click');
    expect(scrollSpy).toHaveBeenCalled();
  });
});
