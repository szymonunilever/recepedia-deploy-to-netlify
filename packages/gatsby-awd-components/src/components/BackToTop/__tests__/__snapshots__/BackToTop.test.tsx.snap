// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back to top matches the snapshot 1`] = `
<div
  style={
    Object {
      "fontSize": "2rem",
      "height": "200vh",
    }
  }
>
  <Component
    content={
      Object {
        "label": "Back to top",
      }
    }
    viewType={1}
  >
    <div
      className="back-to-top backToTop"
    >
      <Component
        attributes={
          Object {
            "aria-label": "back to top",
          }
        }
        className="backToTop__button back-to-top__button"
        onClick={[Function]}
        viewType={1}
      >
        <button
          aria-label="back to top"
          className="buttonIcon button--icon backToTop__button back-to-top__button"
          data-componentname="button"
          data-componentview="button"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          type="button"
        />
      </Component>
      <span
        className="back-to-top__label"
      >
        Back to top
      </span>
    </div>
  </Component>
</div>
`;
